OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -51,13 +51,27 @@ public class McvManagerBotModuleInfo : ConditionalTraitInfo
 
 	public class McvManagerBotModule : ConditionalTrait<McvManagerBotModuleInfo>, IBotTick, IBotPositionsUpdated, IGameSaveTraitData
 	{
-		public CPos GetRandomBaseCenter()
+		public CPos GetRandomBaseCenter(bool distanceToBaseIsImportant)
 		{
+			if (distanceToBaseIsImportant == false)
+				return initialBaseCenter;
+
+			var tileset = world.Map.Rules.TileSet;
+			BitArray resourceTypeIndices;
+
+			resourceTypeIndices = new BitArray(tileset.TerrainInfo.Length);
```

Comment:
You can join these two `var resourceTypeIndices = new BitArray(tileset.TerrainInfo.Length);`